File: Post/DrQuine/Lua/Grace.lua
-- This program will print its own source when run.
local Colleen = "-- This program will print its own source when run.%1local Colleen = %q%1%1function Grace()%1\9local file = io.open(%2Grace_kid.lua%2, %2w%2)%1\9if file == nil then return end%1%1\9local str = Colleen:gsub(%2%%%%.%2, {[%2%%1%2] = string.char(10), [%2%%2%2] = '%2'})%1\9file:write(str:format(Colleen))%1\9file:close()%1end%1%1Grace()"

function Grace()
	local file = io.open("Grace_kid.lua", "w")
	if file == nil then return end

	local str = Colleen:gsub("%%.", {["%1"] = string.char(10), ["%2"] = '"'})
	file:write(str:format(Colleen))
	file:close()
end

Grace()

File: Post/DrQuine/C/Grace.c
#include <stdio.h>
#define COLLEEN "#include <stdio.h>%2$c#define COLLEEN %3$c%1$s%3$c%2$c#define GRACE int main() {FILE *f = fopen(%3$cGrace_kid.c%3$c, %3$cw%3$c); if (!f) return 1; fprintf(f, COLLEEN, COLLEEN, 10, 34); fclose(f);}%2$c/*%2$c	This program will print its own source when run.%2$c*/%2$c%2$cGRACE"
#define GRACE int main() {FILE *f = fopen("Grace_kid.c", "w"); if (!f) return 1; fprintf(f, COLLEEN, COLLEEN, 10, 34); fclose(f);}
/*
	This program will print its own source when run.
*/

GRACE